### PR TITLE
add lovasz loss, mixed loss, road extration dataset

### PR DESCRIPTION
**Pull request by LutaoChu**
1 新增lovasz softmax loss和lovasz hinge loss
2 新增MixedLoss类，在不改动组网代码下实现混合loss训练
3 新增mini deep globe road extraction二分类数据集

### CODE REVIEW

File: configs/ocrnet/ocrnet_hrnetw18_cityscapes_1024x512_160k_lovasz_softmax.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+batch_size: 2
+iters: 160000
+
+model:
+  type: OCRNet
+  backbone:
+    type: HRNet_W18
+    pretrained: https://bj.bcebos.com/paddleseg/dygraph/hrnet_w18_ssld.tar.gz
+  num_classes: 19
```
Comment (michaelowenliu) [inline configs/ocrnet/ocrnet_hrnetw18_cityscapes_1024x512_160k_lovasz_softmax.yml]: model的num_classes去掉

File: configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  transforms:
+    - type: Normalize
+  mode: val
+
+model:
+  type: OCRNet
+  backbone:
+    type: HRNet_W18
+    pretrained: https://bj.bcebos.com/paddleseg/dygraph/hrnet_w18_ssld.tar.gz
+  num_classes: 19
```
Comment (michaelowenliu) [inline configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k.yml]: 去掉num_classes

File: configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k_lovasz_hinge.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  transforms:
+    - type: Normalize
+  mode: val
+
+model:
+  type: OCRNet
+  backbone:
+    type: HRNet_W18
+    pretrained: https://bj.bcebos.com/paddleseg/dygraph/hrnet_w18_ssld.tar.gz
+  num_classes: 19
```
Comment (michaelowenliu) [inline configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k_lovasz_hinge.yml]: 去掉num_classes

File: paddleseg/cvlibs/config.py [hunk 4]
```diff
@@ -211,7 +211,9 @@ def loss(self) -> dict:
                 if key == 'types':
                     self._losses['types'] = []
                     for item in args['types']:
-                        item['ignore_index'] = self.train_dataset.ignore_index
+                        if item['type'] != 'MixedLoss':
+                            item[
```
Comment (michaelowenliu) [inline paddleseg/cvlibs/config.py]: 改下格式

File: paddleseg/datasets/mini_deep_globe_road_extraction.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+@manager.DATASETS.add_component
+class MiniDeepGlobeRoadExtraction(Dataset):
+    """
+    MiniDeepGlobeRoadExtraction dataset is extraced from DeepGlobe CVPR2018 challenge (http://deepglobe.org/)
+
+    There are 800 images in the training set and 200 images in the validation set.
+
+    Args:
+        dataset_root (str, optional): The dataset directory. Default: None
```
Comment (michaelowenliu) [inline paddleseg/datasets/mini_deep_globe_road_extraction.py]: 注释最后加句号

File: paddleseg/models/losses/lovasz_loss.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+        super(LovaszSoftmaxLoss, self).__init__()
+        self.ignore_index = ignore_index
+        self.classes = classes
+
+    def forward(self, logits, labels):
+        """
+        Forward computation.
+
+        Args:
+            logits: [N, C, H, W] Tensor, logits at each prediction (between -\infty and +\infty)
```
Comment (michaelowenliu) [inline paddleseg/models/losses/lovasz_loss.py]: Seg注释规范，类型

**Review by michaelowenliu — CHANGES_REQUESTED** [review]
已提review意见

File: configs/ocrnet/ocrnet_hrnetw18_cityscapes_1024x512_160k_lovasz_softmax.yml (hunk 1 above)
Comment (LutaoChu) [reply configs/ocrnet/ocrnet_hrnetw18_cityscapes_1024x512_160k_lovasz_softmax.yml]: done

File: configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k.yml (hunk 2 above)
Comment (LutaoChu) [reply configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k.yml]: done

File: configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k_lovasz_hinge.yml (hunk 3 above)
Comment (LutaoChu) [reply configs/ocrnet/ocrnet_hrnetw18_road_extraction_768x768_15k_lovasz_hinge.yml]: done

File: paddleseg/cvlibs/config.py (hunk 4 above)
Comment (LutaoChu) [reply paddleseg/cvlibs/config.py]: done

File: paddleseg/datasets/mini_deep_globe_road_extraction.py (hunk 5 above)
Comment (LutaoChu) [reply paddleseg/datasets/mini_deep_globe_road_extraction.py]: done

File: paddleseg/models/losses/lovasz_loss.py (hunk 6 above)
Comment (LutaoChu) [reply paddleseg/models/losses/lovasz_loss.py]: done

**Review by michaelowenliu — APPROVED** [review]
